signer/src/api/new_block.rs

```diff
@@ -112,34 +143,266 @@ pub async fn new_block_handler(state: State<ApiState<impl Context>>, body: Strin
+///
+/// # Parameters
+/// - `ctx`: Shared application context with configuration and database access.
+/// - `event`: The withdrawal acceptance event to be processed.
+/// - `stacks_chaintip`: Current Stacks blockchain chaintip information for
+///   context on block height and hash.
+///
+/// # Returns
+/// - `Result<WithdrawalUpdate, Error>`: On success, returns a `WithdrawalUpdate` struct
+///   for Emily containing relevant withdrawal information. Returns an `Error` if a database
+///   operation fails or if Bitcoin block information is missing.
+async fn handle_withdrawal_accept(
+    ctx: &impl Context,
+    event: WithdrawalAcceptEvent,
+    stacks_chaintip: &StacksBlock,
+) -> Result<WithdrawalUpdate, Error> {
+    ctx.get_storage_mut()
+        .write_withdrawal_accept_event(&event)
+        .await?;
+
+    let bitcoin_block =
+        fetch_bitcoin_block_from_txid(&ctx.get_storage(), event.outpoint.txid).await?;
+    Ok(WithdrawalUpdate {
+        request_id: event.request_id,
+        status: Status::Confirmed,
+        fulfillment: Some(Some(Box::new(Fulfillment {
+            bitcoin_block_hash: bitcoin_block.block_hash.to_string(),
+            bitcoin_block_height: bitcoin_block.block_height,
+            bitcoin_tx_index: event.outpoint.vout,
+            bitcoin_txid: event.outpoint.txid.to_string(),
+            btc_fee: event.fee,
+            stacks_txid: event.txid.to_hex(),
+        }))),
+        status_message: format!("Included in block {}", event.block_id.to_hex()),
+        last_update_block_hash: stacks_chaintip.block_hash.to_hex(),
+        last_update_height: stacks_chaintip.block_height,
+    })
+}
+
+/// Processes a withdrawal creation event, adding new withdrawal records to the
+/// database and preparing the data for Emily.
+///
+/// # Parameters
+/// - `ctx`: Shared application context containing configuration and database access.
+/// - `event`: The withdrawal creation event to be processed.
+///
+/// # Returns
+/// - `Result<CreateWithdrawalRequestBody, Error>`: On success, returns a `CreateWithdrawalRequestBody`
+///   with withdrawal information. If database access fails or Stacks block data is missing,
+///   returns an `Error`.
+async fn handle_withdrawal_create(
+    ctx: &impl Context,
+    event: WithdrawalCreateEvent,
+) -> Result<CreateWithdrawalRequestBody, Error> {
+    ctx.get_storage_mut()
+        .write_withdrawal_create_event(&event)
+        .await?;
+
+    let stacks_block = ctx
+        .get_storage()
+        .get_stacks_block(&StacksBlockHash::from(event.block_id))
+        .await?
+        .ok_or(Error::MissingBlock)?;
+
+    Ok(CreateWithdrawalRequestBody {
+        amount: event.amount,
+        parameters: Box::new(WithdrawalParameters { max_fee: event.max_fee }),
+        recipient: event.recipient.to_string(),
+        request_id: event.request_id,
+        stacks_block_hash: event.block_id.to_hex(),
+        stacks_block_height: stacks_block.block_height,
```

Comment:
The height of the stacks block is given in the webhook body as the `block_height` field. We should use that instead.

Reply:
You are definitely right, unfortunately though for most of the events (3 out of 4) we are missing some values. For this case in particular, I saw the `block_height` field, but it's described to be block height of the bitcoin chain, not for stacks. But maybe the comments are just wrong?

https://github.com/stacks-network/sbtc/blob/main/signer/src/stacks/events.rs#L164

```
    /// The block height of the bitcoin blockchain when the stacks
    /// transaction that emitted this event was executed.
    pub block_height: u64,
    ```

Reply:
Ahh, there are two such fields. I was thinking that you use this one https://github.com/stacks-network/sbtc/blob/7fd644b081f0ece1c241ea5e2b2977f011cfbb64/signer/src/stacks/webhooks.rs#L44-L50

Reply:
Oh got it! About that field, I didn't use it on purpose for a similar reason I explained in the other comment. 
In  this case the `CreateWithdrawalRequestBody` wants the `block_height` and `block_hash` of the `The stacks block hash in which this request id was initiated.`

The one we have in the `NewBlockEvent` is the Stacks chaintip. It should represent the block that confirmed the transaction, not the one that initiated the request, right?

If we can guarantee that the block in the `NewBlockEvent` is the same as the one that originated the request, then I can switch to that!

Reply:
> The one we have in the `NewBlockEvent` is the Stacks chaintip. It should represent the block that confirmed the transaction, not the one that initiated the request, right?

Great question. In this case, the block where the request was initiated is the same as the block where the transaction gets confirmed.

Reply:
Perfect!